components/batch-system/src/router.rs

```diff
@@ -132,19 +153,25 @@ where
     /// Register a mailbox with given address.
     pub fn register(&self, addr: u64, mailbox: BasicMailbox<N>) {
         let mut normals = self.normals.lock().unwrap();
-        if let Some(mailbox) = normals.insert(addr, mailbox) {
+        if let Some(mailbox) = normals.map.insert(addr, mailbox) {
             mailbox.close();
         }
+        normals
+            .alive_cnt
+            .store(normals.map.len(), Ordering::Relaxed);
```

Comment:
Is it possible to call `register` in threads concurrently? If so, `alive_cnt.fetch_add` is better.

Reply:
Which operation has better performance? 🤔

Reply:
I think the problem is the atomic issue about the two operators `map insert` and `map len`.